packages/flutter_tools/lib/src/version.dart

```diff
@@ -488,12 +502,14 @@ class FlutterVersion {
 
     // Cache is empty or it's been a while since the last server ping. Ping the server.
     try {
+      final String remoteFrameworkCommitRevision = await fetchRemoteFrameworkLatestRevision(channel);
       final DateTime remoteFrameworkCommitDate = DateTime.parse(
-        await FlutterVersion.fetchRemoteFrameworkCommitDate(channel),
+       _latestGitCommitDate(branch: remoteFrameworkCommitRevision),
       );
       await versionCheckStamp.store(
         newTimeVersionWasChecked: _clock.now(),
         newKnownRemoteVersion: remoteFrameworkCommitDate,
+        newKnownRemoteRevision: remoteFrameworkCommitRevision,
```

Comment:
Even after `__flutter_version_check__` remote is removed, `git` will keep the dangling fetched commit for [2 weeks](https://git-scm.com/docs/git-gc#Documentation/git-gc.txt---pruneltdategt). This should be okay as the stamp is updated every 3 days, but might fail if the user forcibly prunes using something like `git gc --prune=now`.